agent/rpm/pbench-agent.spec.j2

```diff
@@ -68,19 +59,23 @@ cp -a agent/* %{?buildroot}/%{installdir}/
 %pre
 # this RPM conflicts with a configtools RPM, but we may have a Pypi
 # configtools installed: zap it.
-%if 0%{?rhel} == 7
+%if 0%{?rhel} == 7 && 0%{?rhel_minor} < 7
 scl enable rh-python36 -- bash -c 'if pip3 show configtools > /dev/null 2>&1 ;then pip3 uninstall -y configtools ;fi'
 %endif
 
-%if 0%{?rhel} == 8 || 0%{?fedora} >= 29
+%if 0%{?fedora} >= 31 || 0%{?rhel} == 8 || ( 0%{?rhel} == 7 && 0%{?rhel_minor} >= 7 )
 if pip3 show configtools > /dev/null 2>&1 ;then pip3 uninstall -y configtools ;fi
 %endif
 
 %post
 
 # Install python dependencies
-echo "pip3 install -r /%{installdir}/requirements.txt"
+%if 0%{?rhel} == 7 && 0%{?rhel_minor} < 7
```

Comment:
In the %pre section, we can use tests in the shell - the problem is the Requires: lines.